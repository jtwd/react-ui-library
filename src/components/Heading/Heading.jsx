import React from 'react'
import { oneOf, node, bool } from 'prop-types'

import { H2, H3, H4, H5 } from './Heading.styles'

/** Simple shell for using headings. Level 1 is a H2, level 2 a H3 and so on. H1 is reserved for the PageHeader component */
function Heading({level, children, collapse}) {
  if (!children) return null
  // Allow number strings
  switch(parseInt(level, 10)) {
    case 2:
      return <H3 collapse={collapse}>{children}</H3>
    case 3:
      return <H4 collapse={collapse}>{children}</H4>
    case 4:
      return <H5 collapse={collapse}>{children}</H5>
    default:
      return <H2 collapse={collapse}>{children}</H2>
  }
}

Heading.propTypes = {
  /** Level of heading (1-4 = h2-h5) */
  level: oneOf([1, 2, 3, 4, '1', '2', '3', '4']),
  /** Remove margin from heading */
  collapse: bool,
  /** Content of heading */
  children: node
}

Heading.defaultProps = {
  level: 1,
  children: null,
  collapse: false
}

export default Heading
