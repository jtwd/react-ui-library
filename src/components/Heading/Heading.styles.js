import styled from 'styled-components'
import { white } from '../_theme/colors'
import { fsMd, fsLg, fsXl, fsXxl } from '../_theme/fontSizes'
import { md, sm, xs } from '../_theme/spacers'
import trimChildren from '../_theme/mixins/trimChildren'

const commonStyles = `
  color: ${white};
  ${trimChildren('ver')}
`

export const H2 = styled.h2`
  ${commonStyles}
  margin: ${md} 0;
  font-size: ${fsXxl};
`

export const H3 = styled.h3`
  ${commonStyles}
  margin: ${sm} 0;
  font-size: ${fsXl};
`

export const H4 = styled.h4`
  ${commonStyles}
  margin: ${sm} 0;
  font-size: ${fsLg};
`

export const H5 = styled.h5`
  ${commonStyles}
  font-size: ${fsMd};
  margin: ${xs} 0;
`
