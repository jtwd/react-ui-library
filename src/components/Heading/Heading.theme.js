import { getTheme } from '../_theme/utils'

const { fontSizes, spacing, fonts } = getTheme()

export default {
  default: {
    fontFamily: 'inherit',
    color: 'inherit',
    fontWeight: fonts.primary.bold,
    h2FontSize: fontSizes.xxl,
    h2Margin: `${spacing.md} 0`,
    h3FontSize: fontSizes.xl,
    h3Margin: `${spacing.sm} 0`,
    h4FontSize: fontSizes.lg,
    h4Margin: `${spacing.sm} 0`,
    h5FontSize: fontSizes.md,
    h5Margin: `${spacing.xs} 0`,
    h6FontSize: fontSizes.md,
    h6Margin: `${spacing.xs} 0`,
    h6TextTransform: 'uppercase',
    h6FontWeight: 'normal'
  }
}