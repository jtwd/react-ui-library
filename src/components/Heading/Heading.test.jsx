import React from 'react'
import renderer from 'react-test-renderer'
import Heading from './Heading'

test('Heading renders top level correctly', () => {
  const component = renderer.create(<Heading level={1}>Top level heading</Heading>)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Heading renders second level correctly', () => {
  const component = renderer.create(<Heading level={2}>Second level heading</Heading>)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Heading renders third level correctly', () => {
  const component = renderer.create(<Heading level={3}>Third level heading</Heading>)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Heading renders forth level correctly', () => {
  const component = renderer.create(<Heading level={4}>Forth level heading</Heading>)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Heading renders top level when not passed a level prop', () => {
  const component = renderer.create(<Heading>Top level heading</Heading>)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Heading does not render when not passed any children' , () => {
  const component = renderer.create(<Heading/>)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
