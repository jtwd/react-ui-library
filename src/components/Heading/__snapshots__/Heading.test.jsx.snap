// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading does not render when not passed any children 1`] = `null`;

exports[`Heading renders forth level correctly 1`] = `
<h5
  className="sc-bxivhb bKnpar"
>
  Forth level heading
</h5>
`;

exports[`Heading renders second level correctly 1`] = `
<h3
  className="sc-bwzfXH gNUacx"
>
  Second level heading
</h3>
`;

exports[`Heading renders third level correctly 1`] = `
<h4
  className="sc-htpNat klDYeK"
>
  Third level heading
</h4>
`;

exports[`Heading renders top level correctly 1`] = `
<h2
  className="sc-bdVaJa cXeRcu"
>
  Top level heading
</h2>
`;

exports[`Heading renders top level when not passed a level prop 1`] = `
<h2
  className="sc-bdVaJa cXeRcu"
>
  Top level heading
</h2>
`;
